File: bms/go/go-gt/heap_less/heap_less.go
package test

var ()

const ()

func (h Heap) less(a, b int64) bool {
	i, j := h.I[a], h.I[b]
	return h.W[i] < h.W[j]
}

type Heap struct {
	I []int64

	W []int64
}

File: translations_only/fail/go/go-gt/heap_less/mistral/CAPR/heap_less.rs


use std::boxed::Box;

struct Heap {
 i: Vec<usize>,
 w: Vec<i64>,
}

fn less(h: &Heap, a: i64, b: i64) -> bool {
 h.w[h.i[a as usize]] < h.w[h.i[b as usize]]
}

